src/main/java/org/mybatis/caches/redis/SerializeUtil.java

```diff
@@ -19,41 +19,30 @@
 
 public final class SerializeUtil {
 
+  private static Serializer serializer;
+
+  static {
+    RedisConfig redisConfig = RedisConfigurationBuilder.getInstance().parseConfiguration();
+    if ("kryo".equals(redisConfig.getSerializer())) {
+      serializer = KryoSerializer.INSTANCE;
+    } else {
+      serializer = JDKSerializer.INSTANCE;
+    }
+  }
+
   private SerializeUtil() {
     // prevent instantiation
   }
 
   public static byte[] serialize(Object object) {
-    try {
-      //use kryo serialize first
-      return KryoSerializer.serialize(object);
-    } catch (Exception e) {
-      //if kryo serialize fails, user jdk serialize as a fallback
-      try {
-        return JDKSerializer.serialize(object);
-      } catch (CacheException cacheException) {
-        throw cacheException;
-      }
-
-    }
+    return serializer.serialize(object);
   }
 
   public static Object unserialize(byte[] bytes) {
-    if (bytes == null) {
+    if (bytes == null || bytes.length == 0) {
```

Comment:
Hi @laddcn ,
Just curious, in what scenario does this method receive an empty byte array?

Reply:
I add this just for protection.  Since this method is public, some one may use this method in a wrong way in the future.

Reply:
Ah, I see.
In that case, it probably is better to let the serializer throw exception.
I'm making a small change, so I'll take care of it.
Thanks for the swift response!

Reply:
@harawata Thank you for suggestion. 😃